awx/main/tasks.py

```diff
@@ -1563,46 +1569,118 @@ def should_use_proot(self, job):
         '''
         return getattr(settings, 'AWX_PROOT_ENABLED', False)
 
-    def pre_run_hook(self, job):
+    def pre_run_hook(self, job, private_data_dir):
         if job.inventory is None:
             error = _('Job could not start because it does not have a valid inventory.')
             self.update_model(job.pk, status='failed', job_explanation=error)
             raise RuntimeError(error)
-        if job.project and job.project.scm_type:
+        elif job.project is None:
+            error = _('Job could not start because it does not have a valid project.')
+            self.update_model(job.pk, status='failed', job_explanation=error)
+            raise RuntimeError(error)
+        elif job.project.status in ('error', 'failed'):
+            msg = _(
+                'The project revision for this job template is unknown due to a failed update.'
+            )
+            job = self.update_model(job.pk, status='failed', job_explanation=msg)
+            raise RuntimeError(msg)
+
+        project_path = job.project.get_project_path(check_if_exists=False)
+        job_revision = job.project.scm_revision
+        needs_sync = True
+        if not job.project.scm_type:
+            # manual projects are not synced, user has responsibility for that
+            needs_sync = False
+        elif not os.path.exists(project_path):
+            logger.debug('Performing fresh clone of {} on this instance.'.format(job.project))
```

Comment:
These added logging statements here all seem like they should probably be info-level, not debug.

Reply:
https://github.com/ansible/awx/pull/3882/files#diff-8f5e2fe91e83c971fc84cb35476362db

A lot of the INFO-level logs got axed en masse (almost all, really), as per the reasons there. I raised the level of some select log statements in this PR. For "Skipping project sync", I can buy INFO, because it explains the lack of a unified job. But for statements like this one, they fit in the category of normal cluster operation and that seems to fit in the DEBUG stream precedent.

Reply:
I agree `DEBUG` is more appropriate here - this is just informational regarding work we're doing, and it's not an abnormal thing to have happen.  Raising the log level will probably make the logs pretty noisy.